### PR TITLE
scfg.0.1

### PR DESCRIPTION
**Pull request by zapashcanon**
Hi, this is the first scfg release.

### CODE REVIEW

File: packages/scfg/scfg.0.1/opam [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+maintainer: ["Léo Andrès <contact@ndrs.fr>"]
+authors: ["Léo Andrès <contact@ndrs.fr>"]
+license: "ISC"
+tags: ["scfg" "configuration" "format" "simple" "config" "parser" "printer"]
+homepage: "https://git.zapashcanon.fr/zapashcanon/scfg"
+doc: "https://doc.zapashcanon.fr/scfg"
+bug-reports: "https://git.zapashcanon.fr/zapashcanon/scfg/issues"
+depends: [
+  "dune" {>= "2.8"}
+  "ocaml" {>= "4.08"}
```
Comment (kit-ty-kate) [inline packages/scfg/scfg.0.1/opam:15]:
```suggestion
  "ocaml" {>= "4.13"}
```
```
#=== ERROR while compiling scfg.0.1 ===========================================#
# context              2.0.10 | linux/x86_64 | ocaml-base-compiler.4.12.1 | pinned(https://git.zapashcanon.fr/zapashcanon/scfg/archive/0.1.tar.gz)
# path                 ~/.opam/4.12/.opam-switch/build/scfg.0.1
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p scfg -j 31 @install
# exit-code            1
# env-file             ~/.opam/log/scfg-8009-60459a.env
# output-file          ~/.opam/log/scfg-8009-60459a.out
### output ###
#       ocamlc src/.scfg.objs/byte/scfg__Pp.{cmi,cmo,cmt} (exit 2)
# (cd _build/default && /home/opam/.opam/4.12/bin/ocamlc.opt -w -40 -g -bin-annot -I src/.scfg.objs/byte -I src/.scfg.objs/public_cmi -I /home/opam/.opam/4.12/lib/menhirLib -I /home/opam/.opam/4.12/lib/sedlex -no-alias-deps -open Scfg -o src/.scfg.objs/byte/scfg__Pp.cmo -c -impl src/pp.pp.ml)
# File "src/pp.ml", line 25, characters 12-25:
# 25 |     else if String.exists (Hashtbl.mem chars_to_quote) param then begin
#                  ^^^^^^^^^^^^^
# Error: Unbound value String.exists
```

File: packages/scfg/scfg.0.1/opam [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+authors: ["Léo Andrès <contact@ndrs.fr>"]
+license: "ISC"
+tags: ["scfg" "configuration" "format" "simple" "config" "parser" "printer"]
+homepage: "https://git.zapashcanon.fr/zapashcanon/scfg"
+doc: "https://doc.zapashcanon.fr/scfg"
+bug-reports: "https://git.zapashcanon.fr/zapashcanon/scfg/issues"
+depends: [
+  "dune" {>= "2.8"}
+  "ocaml" {>= "4.13"}
+  "menhir"
```
Comment (kit-ty-kate) [inline packages/scfg/scfg.0.1/opam:18]: A lower-bound constraint is missing here:
```
# (cd _build/default && /home/opam/.opam/4.13/bin/menhir src/menhir_parser.mly --base src/menhir_parser --infer-write-query src/menhir_parser__mock.ml.mock)
# /home/opam/.opam/4.13/bin/menhir: unknown option '--infer-write-query'.
# Usage: /home/opam/.opam/4.13/bin/menhir <options> <filenames>
#   --base <basename>           Specifies a base name for the output file(s)
#   --canonical                 Construct a canonical Knuth LR(1) automaton
#   --comment                   Include comments in the generated code
#   --compare-errors <filename> (used twice) Compare two .messages files.
#   --compile-errors <filename> Compile a .messages file to OCaml code.
#   --coq                       Generate a formally verified parser, in Coq
#   --coq-no-complete           Do not generate a proof of completeness
#   --coq-no-actions            Ignore semantic actions in the Coq output
#   --depend                    Invoke ocamldep and display dependencies
#   --dump                      Describe the automaton in <basename>.automaton
#   --echo-errors <filename>    Echo the sentences in a .messages file
#   --error-recovery            (no longer supported)
#   --explain                   Explain conflicts in <basename>.conflicts
#   --external-tokens <module>  Import token type definition from <module>
#   --fixed-exception           Declares Error = Parsing.Parse_error
#   --follow-construction       (undocumented)
#   --graph                     Write grammar's dependency graph to <basename>.dot
#   --infer                     Invoke ocamlc for ahead of time type inference
#   --inspection                Generate the inspection API (requires --table)
#   --interpret                 Interpret the sentences provided on stdin
#   --interpret-show-cst        Show a concrete syntax tree upon acceptance
#   --interpret-error           Interpret one sentence that should end in an error
#   --lalr                      Construct an LALR(1) automaton
#   --list-errors               Produce a list of erroneous inputs
#   --log-automaton <level>     Log information about the automaton
#   --log-code <level>          Log information about the generated code
#   --log-grammar <level>       Log information about the grammar
#   --no-code-inlining          (undocumented)
#   --no-inline                 Ignore the %inline keyword.
#   --no-pager                  (undocumented)
#   --no-prefix                 (undocumented)
#   --no-stdlib                 Do not load the standard library
#   --ocamlc <command>          Specifies how ocamlc should be invoked
#   --ocamldep <command>        Specifies how ocamldep should be invoked
#   --only-preprocess           Print grammar and exit
#   --only-preprocess-u         Print grammar with unit actions and exit
#   --only-preprocess-uu        Print grammar with unit actions & tokens and exit
#   --only-tokens               Generate token type definition only, no code
#   --raw-depend                Invoke ocamldep and echo its raw output
#   --stdlib <directory>        Specify where the standard library lies
#   --strict                    Warnings about the grammar are errors
#   --suggest-comp-flags        Suggest compilation flags for ocaml{c,opt}
#   --suggest-link-flags-byte   Suggest link flags for ocamlc
#   --suggest-link-flags-opt    Suggest link flags for ocamlopt
#   --suggest-menhirLib         Suggest where is MenhirLib
#   --suggest-ocamlfind         Show whether Menhir was installed using ocamlfind
#   --table                     Use the table-based back-end
#   --timings                   Display internal timings
#   --trace                     Include tracing instructions in the generated code
#   --unused-token <token>      Do not warn that <token> is unused
#   --unused-tokens             Do not warn about any unused token
#   --update-errors <filename>  Update auto-comments in a .messages file
#   --version                   Show version number and exit
#   -b <basename>               Synonymous with --base <basename>
#   -lg                         Synonymous with --log-grammar
#   -la                         Synonymous with --log-automaton
#   -lc                         Synonymous with --log-code
#   -t                          Synonymous with --table
#   -v                          Synonymous with --dump --explain
#   -help                       Display this list of options
#   --help                      Display this list of options
```

Comment (zapashcanon) [reply packages/scfg/scfg.0.1/opam]: @kit-ty-kate fixed

**Comment by kit-ty-kate** [comment]
Thanks!

**Comment by zapashcanon** [comment]
Thanks. :)

**Comment by kit-ty-kate** [comment]
Could you return the fix in the opam file upstream?

**Pull request merged**